### PR TITLE
Add trace log for task cancellation

### PR DESCRIPTION
**Pull request by dnhatn**
Adding trace logs to the task cancellation and its tests to debug the test failure in #55875.

Relates #55875

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/TransportCancelTasksAction.java [hunk 1]
```diff
@@ -153,23 +159,24 @@ private void setBanOnNodes(String reason, boolean waitForCompletion, Cancellable
...
             new GroupedActionListener<>(ActionListener.map(listener, r -> null), childNodes.size());
-        final BanParentTaskRequest banRequest = BanParentTaskRequest.createSetBanParentTaskRequest(
-            new TaskId(clusterService.localNode().getId(), task.getId()), reason, waitForCompletion);
+        final BanParentTaskRequest banRequest = BanParentTaskRequest.createSetBanParentTaskRequest(taskId, reason, waitForCompletion);
         for (DiscoveryNode node : childNodes) {
             transportService.sendRequest(node, BAN_PARENT_ACTION_NAME, banRequest,
                 new EmptyTransportResponseHandler(ThreadPool.Names.SAME) {
                     @Override
                     public void handleResponse(TransportResponse.Empty response) {
+                        logger.warn("sent ban for tasks with the parent [{}] to the node [{}]", taskId, node);
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/TransportCancelTasksAction.java:172]: should this be trace?

**Review by ywelsch — APPROVED** [review]
One comment, o.w. looks good

**Comment by dnhatn** [comment]
> AutoDateHistogramAggregatorTests.testUnmappedMissing

@elasticmachine run elasticsearch-ci/1

**Comment by dnhatn** [comment]
Thanks Yannick!

**Pull request merged**